packages/roosterjs-editor-dom/lib/utils/getEffectiveBackgroundColor.ts

```diff
@@ -0,0 +1,19 @@
+import findClosestElementAncestor from './findClosestElementAncestor';
+import getComputedStyles from './getComputedStyles';
+
+/**
+ * Get effective background value based on parent nodes
+ * @param node The node to get computed styles from
+ * @param rootNode Top most ancestor node to stop searching for inherited background value
+ * @returns An array of the computed styles
+ */
+export default function getEffectiveBackgroundColor(node: Node, rootNode: Node): string {
```

Comment:
A little bit concern about the performance here since it can recursively call until it get a valid color or hit the root of editor. This function is frequently called especially when user is typing. So we want to make this result to be fast enough, even sometimes it is not very accurate.

We are working on a new ContentModel approach which can solve this kind of issue and return a more accurate format state.

Reply:
It happens always if element is nested in another one. Do you have a reference to the release plan for the new ContentModel?

Reply:
Content Model is under development now.

If you are blocked by this issue, you can create a plugin to handle BeforeCopy/BeforeCut event to manually assign the color to copied content